lib/pbench/server/api/resources/users_api.py

```diff
@@ -276,10 +263,12 @@ def post(self):
         Required headers include
             Authorization:   Bearer <Pbench authentication token (user received upon login)>
```

Comment:
Does the logout code behave well if it is invoked with an expired but otherwise valid token?

My instinct is that we shouldn't be too picky about the supplied token:  all that we really want to do here is to delete it from our database, so if it was in the database and we managed to delete it, I'd say that we were successful, regardless of the current validity of the token.  If we first validate the token (other than by trying to delete it), then I think we create more problems than we solve (e.g., if it is expired, we're presumably going to return an error to the client and leave the expired token in the database to be cleaned up later...each of which is an undesirable outcome under the circumstances).

Reply:
If client uses an expired token on logout we will return an error, however, if we detect the token is expired we immediately delete it from our database (see line 104 auth.py). Logout endpoint is protected with `login_required` decorator therefore if client uses the expired token we won't even proceed with a request but at the same time it will also get deleted from the database (Same for any endpoint that is strictly protected). 

However, I have already recorded this as an issue to track https://github.com/distributed-system-analysis/pbench/issues/2130.